src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/OffenceCodes.kt

```diff
@@ -0,0 +1,100 @@
+  YOI_55_28(paragraph = "28", uniqueOffenceCodes = listOf(24101), paragraphDescription = Descriptions.YOI_28),
+  YOI_55_19(paragraph = "19", withOthers = "55:29I", uniqueOffenceCodes = listOf(17001), paragraphDescription = Descriptions.YOI_19),
+  YOI_55_18(paragraph = "18", uniqueOffenceCodes = listOf(17002), paragraphDescription = Descriptions.YOI_18),
+  YOI_55_21B(paragraph = "21", withOthers = "55:29AE", uniqueOffenceCodes = listOf(19001), paragraphDescription = Descriptions.YOI_21),
+  YOI_55_21C(paragraph = "21", withOthers = "55:29AF", uniqueOffenceCodes = listOf(19002), paragraphDescription = Descriptions.YOI_21),
+  YOI_55_21A(paragraph = "21", withOthers = "55:29AD", uniqueOffenceCodes = listOf(19003), paragraphDescription = Descriptions.YOI_21),
+  YOI_55_23(paragraph = "23", uniqueOffenceCodes = listOf(20001), paragraphDescription = Descriptions.YOI_23_ADULT_20A),
+  YOI_55_20(paragraph = "22", withOthers = "55:29AC", uniqueOffenceCodes = listOf(20002), paragraphDescription = Descriptions.YOI_22_ADULT_20),
+  YOI_55_25(paragraph = "25", withOthers = "55:29W", uniqueOffenceCodes = listOf(22001), paragraphDescription = Descriptions.YOI_25_ADULT_22),
+  YOI_55_26(paragraph = "26", withOthers = "55:29AI", uniqueOffenceCodes = listOf(23101), paragraphDescription = Descriptions.YOI_26_ADULT_23),
+  YOI_55_3A(paragraph = "3", withOthers = "55:29K", uniqueOffenceCodes = listOf(2001, 2021), paragraphDescription = Descriptions.YOI_3_ADULT_2),
+  YOI_55_3C(paragraph = "3", withOthers = "55:29M", uniqueOffenceCodes = listOf(2002), paragraphDescription = Descriptions.YOI_3_ADULT_2),
+  YOI_55_3D(paragraph = "3", withOthers = "55:29N", uniqueOffenceCodes = listOf(2003), paragraphDescription = Descriptions.YOI_3_ADULT_2),
+  YOI_55_3B(paragraph = "3", withOthers = "55:29L", uniqueOffenceCodes = listOf(2004), paragraphDescription = Descriptions.YOI_3_ADULT_2),
+  YOI_55_4(paragraph = "4", withOthers = "55:29H", uniqueOffenceCodes = listOf(3001), paragraphDescription = Descriptions.YOI_4),
+  YOI_55_7(paragraph = "7", withOthers = "55:29R", uniqueOffenceCodes = listOf(6001), paragraphDescription = Descriptions.YOI_7),
+  YOI_55_20A(paragraph = "20", withOthers = "55:29AB", uniqueOffenceCodes = listOf(18001), paragraphDescription = Descriptions.YOI_20),
+  YOI_55_20B(paragraph = "20", withOthers = "55:29AO", uniqueOffenceCodes = listOf(18002), paragraphDescription = Descriptions.YOI_20),
+  YOI_55_24(paragraph = "24", withOthers = "55:29AL", uniqueOffenceCodes = listOf(21001), paragraphDescription = Descriptions.YOI_24_ADULT_21),
+  DEFAULT(paragraph = "", uniqueOffenceCodes = emptyList(), paragraphDescription = Descriptions.DEFAULT);
+
+  fun getNomisCode(isYouthOffender: Boolean) =
+    if (isYouthOffender) this.name.substringAfter("YOI_").replace("_", ":")
+    else this.name.substringAfter("ADULT_").replace("_", ":")
+  fun getNomisCodeWithOthers(isYouthOffender: Boolean) = this.withOthers ?: getNomisCode(isYouthOffender)
```

Comment:
Minor one - for readability could put a new line between this and `getNomisCode` as it's a single expression function and without curly braces they're a bit hard to differentiate